hack/supported_releases.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+DIR="$(realpath "${DIR}")"
+
+ROOT_DIR="$(realpath "${DIR}/..")"
+
+record="${ROOT_DIR}/supported_releases.txt"
+
+function latest_releases() {
+  local ref_image=kindest/node
+  token=$(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:${ref_image}:pull" | jq -r '.token')
+  tags=$(curl "https://registry-1.docker.io/v2/${ref_image}/tags/list" -H "Authorization: Bearer ${token}" | jq -r '.tags | .[]')
+  echo "${tags}" | grep -e '^v\d\+\.\d\+\.\d\+$'
+}
+
+function record_releases() {
+  cat "${record}"
+}
+
+function bump() {
+  local raw_all_releases
+  raw_all_releases="$*"
+  declare -A release_map=()
+  for release in ${raw_all_releases}; do
+    release="${release//v/}"
+    minor="${release%.*}"
+    patch="${release##*.}"
+
+    if [[ "${release_map["${minor}"]}" -le "${patch}" ]]; then
+      release_map["${minor}"]="${patch}"
+    fi
+  done
+
+  for r in "${!release_map[@]}"; do
+    echo "${r}.${release_map[${r}]}"
+  done
+}
+
+function send_pr() {
+  local status
+  local branch
+  local body
+  local latest
+  status=$(git status -s)
+  if [[ "${status}" == "" ]]; then
+    echo "No modification"
+    return 1
+  fi
+  if [[ "${status}" != " M supported_releases.txt" ]]; then
+    echo "Modified unintended documents"
+    echo "${status}"
+    return 1
+  fi
+
+  # Use the latest release as the key to prevent duplicate PRs from being created
+  latest="$(cat supported_releases.txt | head -n 1)"
```

Comment:
this seems hacky, what if a previous release was updated?

i.e. latest release 1.25 did not change but 1.24 was updated?

maybe use the timestamp here as the key to prevent duplicate?

Reply:
If we use the timestamp as the key, then the PR created on the first day without processing will be created again the next day, 
or just use a fixed branch, so that no matter what, it will not be created again.